### PR TITLE
Added Status and $Status operation to CapabilityStatement

### PR DESCRIPTION
**Pull request by PTaladay**
## Description
Updated capability statement to display search parameter status in the documentation field. $status operation is also displayed only when feature flag is enabled. 

I used the documentation field because the firely sdk doesn't allow extending on the search param component or any component that relates to the search parameter.

## Related issues
Addresses [issue AB#10442].

## Testing
Ran through running the metadata endpoint with the different configuration options. 

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Shared.Api/Controllers/OperationDefinitionController.cs [hunk 1]
```diff
@@ -33,19 +33,23 @@ public class OperationDefinitionController : Controller
...
         private readonly OperationsConfiguration _operationConfiguration;
         private readonly FeatureConfiguration _featureConfiguration;
+        private readonly CoreFeatureConfiguration _coreFeatureConfiguration;
 
         public OperationDefinitionController(
             IMediator mediator,
             IOptions<OperationsConfiguration> operationsConfig,
-            IOptions<FeatureConfiguration> featureConfig)
+            IOptions<FeatureConfiguration> featureConfig,
+            IOptions<CoreFeatureConfiguration> coreFeatureConfig)
```
Comment (abiisnn) [inline src/Microsoft.Health.Fhir.Shared.Api/Controllers/OperationDefinitionController.cs:42]: What's the difference between `featureConfig` and `coreFeatureConfig`?

**Review by abiisnn — COMMENTED** [review]
🍒 

Do you think you can add a test to make sure it's called correctly?

File: src/Microsoft.Health.Fhir.Shared.Api/Controllers/OperationDefinitionController.cs (hunk 1 above)
Comment (PTaladay) [reply src/Microsoft.Health.Fhir.Shared.Api/Controllers/OperationDefinitionController.cs:42]: I think the original intent was to separate fhir spec features vs fhir server features? Either way I needed to add the coreFeatureConfiguration to check the selectablesearchparameter feature flag.

File: src/Microsoft.Health.Fhir.Core/Data/OperationDefinition/search-parameter-status.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "resourceType": "OperationDefinition",
+    "id": "status",
+    "url": "[base]/OperationDefinition/search-parameter-status",
+    "version": "1.0.0",
+    "name": "Selectable search parameters",
+    "status": "active",
+    "kind": "operation",
+    "description": "Allows for disabling of search parameters that are not being utilized in order to save space in the datastore. This operation is asynchronous as defined in the [FHIR Asynchronous Request Pattern](http://hl7.org/fhir/async.html)",
```
Comment (rajithaalurims) [inline src/Microsoft.Health.Fhir.Core/Data/OperationDefinition/search-parameter-status.json:9]: nit: this allows customers to either disable or turn it back on as "supported" from disabled state correct? Can we change the description to add something about updating status to supported as well?

File: src/Microsoft.Health.Fhir.Api/Features/Routing/RouteNames.cs [hunk 3]
```diff
@@ -65,6 +65,8 @@ internal static class RouteNames
 
         internal const string PurgeHistoryDefinition = "PurgeHistoryDefinition";
 
+        internal const string SearchParameterStatusOperationDefinition = "SearchParameterStatusOperationDefinition";
```
Comment (fhibf) [inline src/Microsoft.Health.Fhir.Api/Features/Routing/RouteNames.cs:68]: Low priority: You can replace the static string with nameof(SearchParameterStatusOperationDefinition).

**Review by fhibf — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Api/Features/Routing/RouteNames.cs (hunk 3 above)
Comment (PTaladay) [reply src/Microsoft.Health.Fhir.Api/Features/Routing/RouteNames.cs:68]: I can add that as a fix in another PR.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs [hunk 4]
```diff
@@ -668,9 +668,10 @@ public void Build(ICapabilityStatementBuilder builder)
         {
             EnsureArg.IsNotNull(builder, nameof(builder));
 
-            builder.PopulateDefaultResourceInteractions()
-                .SyncSearchParameters()
-                .AddGlobalSearchParameters()
+            var task = Task.Run(() => builder.PopulateDefaultResourceInteractions()
+                .SyncSearchParametersAsync(CancellationToken.None));
+            task.Wait();
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs:673]: Should we be using: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.runsynchronously?view=net-7.0 here?

**Pull request merged**